assets/js/atomic/blocks/product-elements/add-to-cart/product-types/variable/variation-attributes/attribute-picker.js

```diff
@@ -1,104 +1,104 @@
 /**
  * External dependencies
  */
-import { useState, useEffect, useMemo } from '@wordpress/element';
+import { useState, useEffect, useMemo, useRef } from '@wordpress/element';
+import isShallowEqual from '@wordpress/is-shallow-equal';
 
 /**
  * Internal dependencies
  */
 import AttributeSelectControl from './attribute-select-control';
 import {
-	getVariationsMatchingSelectedAttributes,
-	getSelectControlOptions,
+	getVariationMatchingSelectedAttributes,
+	getActiveSelectControlOptions,
 } from './utils';
 
 /**
  * AttributePicker component.
  *
  * @param {*} props Component props.
  */
-const AttributePicker = ( { attributes, variationAttributes } ) => {
+const AttributePicker = ( {
+	attributes,
+	variationAttributes,
+	setRequestParams,
+} ) => {
+	const attributesRef = useRef( attributes );
+	const variationAttributesRef = useRef( variationAttributes );
 	const [ variationId, setVariationId ] = useState( 0 );
+	const [ selectedAttributes, setSelectedAttributes ] = useState( {} );
 
-	// @todo Support default selected attributes in Variation Picker.
-	const [ selectedAttributes, setSelectedAttributes ] = useState( [] );
+	// Keeps refs in sync.
+	useEffect( () => {
+		if ( ! isShallowEqual( attributesRef.current, attributes ) ) {
```

Comment:
We have a [custom `useShallowEqual` hook](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/main/assets/js/base/hooks/use-shallow-equal.js) that could be used (could condense some of this code). Same for the following effect for `variationAttributesRef`.